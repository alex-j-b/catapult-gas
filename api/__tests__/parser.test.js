const { parseGas10 } = require('../src/parser.js');
const gas10Pg = require('./gas10Pg.json');

describe('Parser tests', () => {
    it('should parse data in appropriate format', () => {
        expect(parseGas10(gas10Pg)).toMatchSnapshot();
    });
});
