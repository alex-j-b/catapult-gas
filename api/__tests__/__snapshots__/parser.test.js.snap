// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser tests should parse data in appropriate format 1`] = `
Array [
  Object {
    "data": Array [
      Object {
        "x": "20-09-2021",
        "y": 0,
      },
      Object {
        "x": "19-09-2021",
        "y": 495686,
      },
      Object {
        "x": "18-09-2021",
        "y": 468478,
      },
      Object {
        "x": "17-09-2021",
        "y": 432215,
      },
      Object {
        "x": "16-09-2021",
        "y": 407038,
      },
      Object {
        "x": "15-09-2021",
        "y": 399147,
      },
      Object {
        "x": "14-09-2021",
        "y": 378107,
      },
      Object {
        "x": "13-09-2021",
        "y": 334569,
      },
      Object {
        "x": "12-09-2021",
        "y": 317297,
      },
      Object {
        "x": "11-09-2021",
        "y": 277687,
      },
    ],
    "id": "all",
    "total": 3510224,
  },
  Object {
    "data": Array [
      Object {
        "x": "20-09-2021",
        "y": 0,
      },
      Object {
        "x": "19-09-2021",
        "y": 37112,
      },
      Object {
        "x": "18-09-2021",
        "y": 32184,
      },
      Object {
        "x": "17-09-2021",
        "y": 29322,
      },
      Object {
        "x": "16-09-2021",
        "y": 29641,
      },
      Object {
        "x": "15-09-2021",
        "y": 29293,
      },
      Object {
        "x": "14-09-2021",
        "y": 26757,
      },
      Object {
        "x": "13-09-2021",
        "y": 26983,
      },
      Object {
        "x": "12-09-2021",
        "y": 26208,
      },
      Object {
        "x": "11-09-2021",
        "y": 22674,
      },
    ],
    "id": "1",
    "total": 260174,
  },
  Object {
    "data": Array [
      Object {
        "x": "20-09-2021",
        "y": 0,
      },
      Object {
        "x": "19-09-2021",
        "y": 71454,
      },
      Object {
        "x": "18-09-2021",
        "y": 63467,
      },
      Object {
        "x": "17-09-2021",
        "y": 57797,
      },
      Object {
        "x": "16-09-2021",
        "y": 53545,
      },
      Object {
        "x": "15-09-2021",
        "y": 56537,
      },
      Object {
        "x": "14-09-2021",
        "y": 51222,
      },
      Object {
        "x": "13-09-2021",
        "y": 46387,
      },
      Object {
        "x": "12-09-2021",
        "y": 44363,
      },
      Object {
        "x": "11-09-2021",
        "y": 37874,
      },
    ],
    "id": "2",
    "total": 482646,
  },
  Object {
    "data": Array [
      Object {
        "x": "20-09-2021",
        "y": 0,
      },
      Object {
        "x": "19-09-2021",
        "y": 95914,
      },
      Object {
        "x": "18-09-2021",
        "y": 93757,
      },
      Object {
        "x": "17-09-2021",
        "y": 86887,
      },
      Object {
        "x": "16-09-2021",
        "y": 77890,
      },
      Object {
        "x": "15-09-2021",
        "y": 78088,
      },
      Object {
        "x": "14-09-2021",
        "y": 76291,
      },
      Object {
        "x": "13-09-2021",
        "y": 67304,
      },
      Object {
        "x": "12-09-2021",
        "y": 60394,
      },
      Object {
        "x": "11-09-2021",
        "y": 55994,
      },
    ],
    "id": "3",
    "total": 692519,
  },
  Object {
    "data": Array [
      Object {
        "x": "20-09-2021",
        "y": 0,
      },
      Object {
        "x": "19-09-2021",
        "y": 129064,
      },
      Object {
        "x": "18-09-2021",
        "y": 122029,
      },
      Object {
        "x": "17-09-2021",
        "y": 114831,
      },
      Object {
        "x": "16-09-2021",
        "y": 112590,
      },
      Object {
        "x": "15-09-2021",
        "y": 102063,
      },
      Object {
        "x": "14-09-2021",
        "y": 103962,
      },
      Object {
        "x": "13-09-2021",
        "y": 81983,
      },
      Object {
        "x": "12-09-2021",
        "y": 77929,
      },
      Object {
        "x": "11-09-2021",
        "y": 73835,
      },
    ],
    "id": "4",
    "total": 918286,
  },
  Object {
    "data": Array [
      Object {
        "x": "20-09-2021",
        "y": 0,
      },
      Object {
        "x": "19-09-2021",
        "y": 162142,
      },
      Object {
        "x": "18-09-2021",
        "y": 157041,
      },
      Object {
        "x": "17-09-2021",
        "y": 143378,
      },
      Object {
        "x": "16-09-2021",
        "y": 133372,
      },
      Object {
        "x": "15-09-2021",
        "y": 133166,
      },
      Object {
        "x": "14-09-2021",
        "y": 119875,
      },
      Object {
        "x": "13-09-2021",
        "y": 111912,
      },
      Object {
        "x": "12-09-2021",
        "y": 108403,
      },
      Object {
        "x": "11-09-2021",
        "y": 87310,
      },
    ],
    "id": "5",
    "total": 1156599,
  },
]
`;
